core/templates/dev/head/domain/story_viewer/story-viewer-backend-api.service.ts

```diff
@@ -16,59 +16,79 @@
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('StoryViewerBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService', 'STORY_DATA_URL_TEMPLATE',
-  'STORY_PROGRESS_URL_TEMPLATE', function(
-      $http, $q, UrlInterpolationService, STORY_DATA_URL_TEMPLATE,
-      STORY_PROGRESS_URL_TEMPLATE) {
-    var storyDataDict = null;
-    var _fetchStoryData = function(storyId, successCallback, errorCallback) {
-      var storyDataUrl = UrlInterpolationService.interpolateUrl(
-        STORY_DATA_URL_TEMPLATE, {
-          story_id: storyId
-        });
+import { StoryViewerDomainConstants } from
+  'domain/story_viewer/story-viewer-domain.constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
 
-      $http.get(storyDataUrl).then(function(response) {
-        storyDataDict = angular.copy(response.data);
-        if (successCallback) {
-          successCallback(storyDataDict);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
-    };
+@Injectable({
+  providedIn: 'root'
+})
+export class StoryViewerBackendApiService {
+  // TODO(#7176): Replace 'any' with the exact type. This has been kept as
+  // 'any' because 'storyDataDict' is a dict with underscore_cased
+  // keys which give tslint errors against underscore_casing in favor of
+  // camelCasing.
+  storyDataDict: any = null;
+
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private http: HttpClient
+  ) {}
 
-    var _recordChapterCompletion = function(
-        storyId, nodeId, successCallback, errorCallback) {
-      var chapterCompletionUrl = UrlInterpolationService.interpolateUrl(
-        STORY_PROGRESS_URL_TEMPLATE, {
-          story_id: storyId,
-          node_id: nodeId
-        });
-      $http.post(chapterCompletionUrl).then(function(response) {
-        successCallback({
-          summaries: response.data.summaries,
-          nextNodeId: response.data.next_node_id,
-          readyForReviewTest: response.data.ready_for_review_test});
+  _fetchStoryData(storyId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    let storyDataUrl = this.urlInterpolationService.interpolateUrl(
+      StoryViewerDomainConstants.STORY_DATA_URL_TEMPLATE, {
+        story_id: storyId
       });
-    };
 
-    return {
-      fetchStoryData: function(storyId) {
-        return $q(function(resolve, reject) {
-          _fetchStoryData(storyId, resolve, reject);
-        });
-      },
-      recordChapterCompletion: function(storyId, nodeId) {
-        return $q(function(resolve, reject) {
-          _recordChapterCompletion(storyId, nodeId, resolve, reject);
-        });
+    this.http.get(storyDataUrl).toPromise().then((data) => {
+      this.storyDataDict = angular.copy(data);
```

Comment:
Please do not use `angular.copy`, it is not supported in angular. Use `clonedeep`:
```
import cloneDeep from 'lodash/cloneDeep';
..
this.storyDataDict = cloneDeep(data);
```

Reply:
Done!